tests/stdlib/tos.nim

```diff
@@ -138,4 +139,53 @@ let tm = fromUnix(0) + 100.microseconds
 writeFile("a", "")
 setLastModificationTime("a", tm)
 echo getLastModificationTime("a") == tm
-removeFile("a")
\ No newline at end of file
+removeFile("a")
+
+block normalizedPath:
+  doAssert normalizedPath("/a/b/../../../foo", '/') == "/foo"
+  doAssert normalizedPath("\\a\\b\\..\\..\\..\\foo", '\\') == "\\foo"
+
+when defined(Linux) or defined(osx):
+
+  block normalizedPath:
+    block relative:
+      doAssert normalizedPath(".") == "."
+      doAssert normalizedPath("..") == ".."
+      doAssert normalizedPath("../") == ".."
+      doAssert normalizedPath("../..") == "../.."
+      doAssert normalizedPath("../a/..") == ".."
+      doAssert normalizedPath("../a/../") == ".."
+      doAssert normalizedPath("./") == "."
+
+    block absolute:
+      doAssert normalizedPath("/") == "/"
+      doAssert normalizedPath("/.") == "/"
+      doAssert normalizedPath("/..") == "/"
+      doAssert normalizedPath("/../") == "/"
+      doAssert normalizedPath("/../..") == "/"
+      doAssert normalizedPath("/../../") == "/"
+      doAssert normalizedPath("/../../../") == "/"
```

Comment:
I wonder if it would make more sense to raise an error for these.

Reply:
That's the point of normalization. If you are looking for a way to detect and prevent directory traversal I think that should go into joinPath